hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -321,6 +321,84 @@ private AbsoluteTableIdentifier getAbsoluteTableIdentifier(Configuration configu
+   * @param job
+   * @param targetSegment
+   * @param oldPartitionIdList  get old partitionId before partitionInfo was changed
+   * @return
+   * @throws IOException
+   */
+  public List<InputSplit> getSplitsOfOneSegment(JobContext job, String targetSegment,
+      List<Integer> oldPartitionIdList, PartitionInfo partitionInfo)
+      throws IOException {
+    AbsoluteTableIdentifier identifier = getAbsoluteTableIdentifier(job.getConfiguration());
+    List<String> invalidSegments = new ArrayList<>();
+    List<UpdateVO> invalidTimestampsList = new ArrayList<>();
+
+    List<String> segmentList = new ArrayList<>();
+    segmentList.add(targetSegment);
+    setSegmentsToAccess(job.getConfiguration(), segmentList);
+    try {
+
+      // process and resolve the expression
+      Expression filter = getFilterPredicates(job.getConfiguration());
+      CarbonTable carbonTable = getOrCreateCarbonTable(job.getConfiguration());
+      // this will be null in case of corrupt schema file.
+      if (null == carbonTable) {
+        throw new IOException("Missing/Corrupt schema file for table.");
+      }
+
+      CarbonInputFormatUtil.processFilterExpression(filter, carbonTable);
+
+      // prune partitions for filter query on partition table
+      String partitionIds = job.getConfiguration().get(ALTER_PARTITION_ID);
+      BitSet matchedPartitions = null;
+      if (partitionInfo != null) {
+        matchedPartitions = setMatchedPartitions(partitionIds, filter, partitionInfo);
+        if (matchedPartitions != null) {
+          if (matchedPartitions.cardinality() == 0) {
+            return new ArrayList<InputSplit>();
+          } else if (matchedPartitions.cardinality() == partitionInfo.getNumPartitions()) {
+            matchedPartitions = null;
+          }
+        }
+      }
+
+      FilterResolverIntf filterInterface = CarbonInputFormatUtil.resolveFilter(filter, identifier);
+      // do block filtering and get split
+      List<InputSplit> splits = getSplits(job, filterInterface, segmentList, matchedPartitions,
+          partitionInfo, oldPartitionIdList);
+      // pass the invalid segment to task side in order to remove index entry in task side
+      if (invalidSegments.size() > 0) {
+        for (InputSplit split : splits) {
+          ((CarbonInputSplit) split).setInvalidSegments(invalidSegments);
+          ((CarbonInputSplit) split).setInvalidTimestampRange(invalidTimestampsList);
+        }
+      }
+      return splits;
+    } catch (IOException e) {
+      throw new RuntimeException("Can't get splits of the target segment ", e);
+    }
+  }
+
+  private BitSet setMatchedPartitions(String partitionIds, Expression filter,
+      PartitionInfo partitionInfo) {
+    BitSet matchedPartitions = null;
+    if (null != partitionIds) {
+      String[] partList = partitionIds.replace("[", "").replace("]", "").split(",");
+      matchedPartitions = new BitSet(Integer.parseInt(partList[0]));
```

Comment:
Currently the partitionIds from alter table statement could only be one element.
For example 
'alter table t0 split(4) into XXX'
'alter table t1 drop partition(3)' (will submit in another PR ) 
We suppose drop partition is a dangerous operation and allow to drop only one partition in a time.
Maybe in the future we can discuss to extend drop multiple partitions in one statement.
And also we can extend MERGE PARTITION action in later versions(no plan in version1.2)